lib/index.js

```diff
@@ -307,28 +307,25 @@ InsightAPI.prototype.blockEventHandler = function(hashBuffer) {
   }
 };
 InsightAPI.prototype.transactionEventHandler = function(txBuffer) {
-  var tx = new Transaction().fromBuffer(txBuffer);
-  var result = this.txController.transformInvTransaction(tx);
-
-  setTimeout(function(result, txBuffer) { // delay tx relay for 900ms to see if we receive transaction lock
-    var hash = dashcore.crypto.Hash.sha256sha256(txBuffer);
-    var id = hash.toString('binary');
-
-    if (this.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
-      result.txlock = true;
-    }
+  const txVersion = txBuffer.slice(0, 1).toString('hex');
+  if (txVersion === '01' || txVersion === '02' || txVersion === '03') {
```

Comment:
I guess this checks helps you detect existing transaction versions, but I'm not sure what exactly it prevents. It somehow related to the trace mentioned in the PR, but what exact data will lead to crash?

Reply:
Yes, what caused the crash was a `txBuffer` with value as hexstring: `6861736874786c6f636b`. Which when you convert to ascii is `hashtxlock`. My apologies for the unclear explanation in the OP :).

Reply:
Thank you for the explanation! I'm a little bit worried though, since we can introduce new transaction types and forgot to change it here. Can we check that the message isn't a `hashtxlock` instead?

Reply:
I think it might be something else, an unsupported transaction for example. What if we just try to parse transaction and if we can't we should log this error it and skip it. https://github.com/dashevo/insight-api/pull/60#discussion_r296186078

Reply:
Fixed.